lib/charms/identity_platform_login_ui/v0/login_ui_endpoint.py

```diff
@@ -0,0 +1,158 @@
+```shell
+cd some-charm
+charmcraft fetch-lib charms.identity_platform_login_ui.v0.login_ui_endpoint
+```
+To use the library from the requirer side:
+In the `metadata.yaml` of the charm, add the following:
+```yaml
+requires:
+  ui-endpoint-info:
+    interface: login_ui_endpoint
+    limit: 1
+```
+Then, to initialise the library:
+```python
+from charms.identity_platform_login_ui.v0.login_ui_endpoint import (
+    LoginUIEndpointRelationError,
+    LoginUIEndpointRequirer,
+)
+Class SomeCharm(CharmBase):
+    def __init__(self, *args):
+        self.login_ui_endpoint_relation = LoginUIEndpointRequirer(self)
+        self.framework.observe(self.on.some_event_emitted, self.some_event_function)
+    def some_event_function():
+        # fetch the relation info
+        try:
+            login_ui_endpoint = self.login_ui_endpoint_relation.get_login_ui_endpoint()
+        except LoginUIEndpointRelationError as error:
+            ...
+```
+"""
+
+import logging
+from typing import Dict, Optional
+
+from ops.charm import CharmBase, RelationCreatedEvent
+from ops.framework import EventBase, EventSource, Object, ObjectEvents
+from ops.model import Application
+
+# The unique Charmhub library identifier, never change it
+LIBID = "placeholder_ID"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 0
+
+RELATION_NAME = "ui-endpoint-info"
+INTERFACE_NAME = "login_ui_endpoint"
+logger = logging.getLogger(__name__)
+
+
+class LoginUIEndpointRelationReadyEvent(EventBase):
+    """Event to notify the charm that the relation is ready."""
+
+
+class LoginUIEndpointProviderEvents(ObjectEvents):
+    """Event descriptor for events raised by `LoginUIEndpointProvider`."""
+
+    ready = EventSource(LoginUIEndpointRelationReadyEvent)
+
+
+class LoginUIEndpointProvider(Object):
+    """Provider side of the endpoint-info relation."""
+
+    on = LoginUIEndpointProviderEvents()
+
+    def __init__(self, charm: CharmBase, relation_name: str = RELATION_NAME):
+        super().__init__(charm, relation_name)
+
+        self._charm = charm
+        self._relation_name = relation_name
+
+        events = self._charm.on[relation_name]
+        self.framework.observe(
+            events.relation_created, self._on_provider_endpoint_relation_created
+        )
+
+    def _on_provider_endpoint_relation_created(self, event: RelationCreatedEvent) -> None:
+        self.on.ready.emit()
+
+    def send_endpoint_relation_data(self, endpoint: str) -> None:
+        """Updates relation with endpoint info."""
+        if not self._charm.unit.is_leader():
+            return
+
+        relations = self.model.relations[self._relation_name]
+        for relation in relations:
+            relation.data[self._charm.app].update(
+                {
+                    "endpoint": endpoint,
```

Comment:
Do we want to provide only the bare address of login ui without specifying its full endpoints?
Hydra and Kratos need the login form and error urls. Right now both construct these endpoints by adding `/login` or `/oidc_error` to the login ui url (see [here](https://github.com/canonical/kratos-operator/blob/main/src/charm.py#L165)), but as `identity-platform-login-ui-operator` implements them, I think it should be able to provide its full endpoints.